### PR TITLE
feat(route):  leetcode.com daily question

### PR DESCRIPTION
**Pull request by IvanWng97**
## 该 PR 相关 Issue / Involved issue

Close #

## 完整路由地址 / Example for the proposed route(s)

```routes
/leetcode/dailyquestion/en
```

## 新 RSS 检查列表 / New RSS Script Checklist
  
- [x] 新的路由 New Route
  - [x] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [x] 文档说明 Documentation
  - [x] 中文文档 CN
  - [X] 英文文档 EN
- [ ] 全文获取 fulltext
  - [ ] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [ ] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [ ] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

### CODE REVIEW

File: lib/v2/leetcode/dailyquestion-en.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        link: '',
+        titleSlug: '',
+        content: '',
+        frontedId: '',
+        difficulty: '',
+        tags: '',
+    };
+    const url = host + '/graphql';
+    const dailyQuestionPayload = {
+        query: '\n    query questionOfToday {\n  activeDailyCodingChallengeQuestion {\n    date\n    userStatus\n    link\n    question {\n      acRate\n      difficulty\n      freqBar\n      frontendQuestionId: questionFrontendId\n      isFavor\n      paidOnly: isPaidOnly\n      status\n      title\n      titleSlug\n      hasVideoSolution\n      hasSolution\n      topicTags {\n        name\n        id\n        slug\n      }\n    }\n  }\n}\n    ',
```
Comment (SukkaW) [inline lib/v2/leetcode/dailyquestion-en.js:19]: Use template literal to write GraphQL.

File: lib/v2/leetcode/dailyquestion-en.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        query: 'query questionData($titleSlug: String!) {\n  question(titleSlug: $titleSlug) {\n    questionId\n    questionFrontendId\n    boundTopicId\n    title\n    titleSlug\n    content\n    translatedTitle\n    translatedContent\n    isPaidOnly\n    difficulty\n    likes\n    dislikes\n    isLiked\n    similarQuestions\n    exampleTestcases\n    categoryTitle\n    contributors {\n      username\n      profileUrl\n      avatarUrl\n      __typename\n    }\n    topicTags {\n      name\n      slug\n      translatedName\n      __typename\n    }\n    companyTagStats\n    codeSnippets {\n      lang\n      langSlug\n      code\n      __typename\n    }\n    stats\n    hints\n    solution {\n      id\n      canSeeDetail\n      paidOnly\n      hasVideoSolution\n      paidOnlyVideo\n      __typename\n    }\n    status\n    sampleTestCase\n    metaData\n    judgerAvailable\n    judgeType\n    mysqlSchemas\n    enableRunCode\n    enableTestMode\n    enableDebugger\n    envInfo\n    libraryUrl\n    adminUrl\n    challengeQuestion {\n      id\n      date\n      incompleteChallengeCount\n      streakCount\n      type\n      __typename\n    }\n    __typename\n  }\n}\n',
```
Comment (SukkaW) [inline lib/v2/leetcode/dailyquestion-en.js:37]: Use template literal to write GraphQL.

**Review by TonyRL — COMMENTED** [review]
Please add en docs

**Comment by NavePnow** [comment]
> Please add en docs

added support for en docs in latest commit

File: docs/en/programming.md [hunk 3]
```diff
@@ -305,6 +305,11 @@ Subscribe to the updates (threads and submission) from a paritcular Hacker News
 
 <RouteEn author="NathanDai" example="/leetcode/submission/us/nathandai" path="/leetcode/submission/:country/:user" :paramsDesc="['country, Chines(cn) and US(us)', 'Username, available at the URL of the LeetCode user homepage']"/>
 
+### Daily Question
+
+<Route author="NavePnow" example="/leetcode/dailyquestion/en" path="/leetcode/dailyquestion/en"/>
+
```
Comment (TonyRL) [inline docs/en/programming.md:311]:
```suggestion
<RouteEn author="NavePnow" example="/leetcode/dailyquestion/en" path="/leetcode/dailyquestion/en"/>
```

**Review by TonyRL — COMMENTED** [review]
With template literals, multi-line strings [do not require `\n`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#multi-line_strings) anymore

**Comment by NavePnow** [comment]
> With template literals, multi-line strings [do not require `\n`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#multi-line_strings) anymore



> With template literals, multi-line strings [do not require `\n`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#multi-line_strings) anymore

changed graphql string in latest commit

**Review by TonyRL — COMMENTED** [review]
If you read the MDN docs carefully, template literal achieves multi-line strings by replacing `\n` with natural line-breaks but not squeezing everything into one line.

**Comment by NavePnow** [comment]
> If you read the MDN docs carefully, template literal achieves multi-line strings by replacing `\n` with natural line-breaks but not squeezing everything into one line.

resolved in the latest commit. Appreciate your help!

**Pull request merged**